src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java

```diff
@@ -106,4 +146,49 @@ public void validateCallbackErrorsAndWarnings(final CallbackResponse callbackRes
                 .withWarnings(callbackResponse.getWarnings());
         }
     }
+
+    private List<CallbackRetryContext> buildCallbackRetryContexts(final List<Integer> callbackRetryTimeouts) {
+        List<CallbackRetryContext> retryContextList = Lists.newArrayList();
+        if (isCallbackRetriesDisabled(callbackRetryTimeouts)) {
+            retryContextList.add(new CallbackRetryContext(0, callbackRetryTimeouts.remove(0)));
+        } else {
+            this.defaultCallbackRetryIntervalsInSeconds.forEach(cbRetryInterval -> {
+                retryContextList.add(new CallbackRetryContext(cbRetryInterval, defaultCallbackTimeoutInMillis));
+            });
+        }
+        return retryContextList;
+    }
+
+    private boolean isCallbackRetriesDisabled(final List<Integer> callbackRetryTimeouts) {
+        return callbackRetryTimeouts.size() == 1 && callbackRetryTimeouts.get(0) == 0;
```

Comment:
we already discussed about this, so you can ignore this comment. Commenting anyway just for future reference.
Why we are mandating that the value must be `0` for disabling retries? That was not a requirement of RDM-4522. And we can achieve disabling retries without forcing the value to `0`. Retries are disabled by passing one single value rather than a list. The values denote the max callback timeout

Reply:
Sure and it was already explained 👍

Reply:
`0` was chosen as after 4522 it would become an invalid case (0 is not a valid callback timeout) and should not be honoured so we can treat it as a missed requirement.
Also allowing value other `0` could confuse ppl using it now (we would have to announce that too that single values other than `0` also mean disable timeouts) which would confuse people even more. This is counterintuitive and would not make sense to services. We would have to then revert it's meaning when 4522 comes in because we would want single non `0` values to now have a callback timeout meaning rather than merely disabling retry timeouts.
This ticket was to introduce an optional disable of retries whilst making the least amount of functional change possible